From e1d21f22fd1158de2c9397d0bec16bfcb892101f Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Mon, 4 Mar 2013 14:01:20 +0100
Subject: [PATCH 511/628] usg: gadget: Simplify driver initialization

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 65ca9b8ef169bff08e2c44dbb01a5d7ecb44cac0

Use module_platform_driver macro which creates
init and exit functions.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/usb/gadget/xilinx_udc.c |   25 +------------------------
 1 files changed, 1 insertions(+), 24 deletions(-)

diff --git a/drivers/usb/gadget/xilinx_udc.c b/drivers/usb/gadget/xilinx_udc.c
index b551923..901d4d8 100644
--- a/drivers/usb/gadget/xilinx_udc.c
+++ b/drivers/usb/gadget/xilinx_udc.c
@@ -2419,30 +2419,7 @@ static struct of_platform_driver usb_of_driver = {
 	.remove = __devexit_p(usb_of_remove),
 };
 
-/**
- * udc_init() - The device driver initialization function.
- *
- * returns: 0 for success and error value on failure
- *
- **/
-static int __init udc_init(void)
-{
-	return platform_driver_register(&usb_of_driver);
-}
-
-/**
- * udc_exit() - The device driver exit function.
- *
- * returns: 0 for success and error value on failure
- *
- **/
-static void __devexit udc_exit(void)
-{
-	platform_driver_unregister(&usb_of_driver);
-}
-
-module_init(udc_init);
-module_exit(udc_exit);
+module_platform_driver(usb_of_driver);
 
 MODULE_DESCRIPTION("Xilinx udc driver");
 MODULE_AUTHOR("Xilinx, Inc");
-- 
1.7.5.4

